jetty-memcached/jetty-memcached-sessions/src/main/java/org/eclipse/jetty/memcached/session/MemcachedSessionDataMap.java

```diff
@@ -155,14 +161,48 @@ public void initialize(SessionContext context)
     @Override
     public SessionData load(String id) throws Exception
     {
-        SessionData data = _client.get(id);
-        return data;
+        if (!isStarted())
+            throw new IllegalStateException("Not started");
+        
+        final FuturePromise<SessionData> result = new FuturePromise<>();
+
+        Runnable r = () ->
+        {
+            try
+            {
+                result.succeeded(_client.get(id));
+            }
+            catch (Exception e)
+            {
+                result.failed(e);
+            }
+        };
+
+        _context.run(r);
```

Comment:
Elsewhere in the codebase we have done something like:
```suggestion
        if (_context == null)
            r.run();
        else
            _context.run(r);
```
I'm not sure if sessions can work without contexts???

Reply:
The `_context` here is not a `ContextHandler` but a `SessionContext` that has a `run()` method that implements this very recipe, see https://github.com/eclipse/jetty.project/blob/jetty-10.0.x/jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionContext.java#L84